### PR TITLE
CLOUDSTACK-10250: Reinstall VM from bypassed registered template

### PR DESCRIPTION
**Pull request by nvazquez**
JIRA Ticket: https://issues.apache.org/jira/browse/CLOUDSTACK-10250

This fix allows users to restore a VM from a previously registered template using the Direct Download option (only for KVM currently)

NOTE: As Reinstall VM button prompts only featured templates, to be able to restore a vm to a Direct Download template, it should be registered as Featured

### CODE REVIEW

**Comment by borisstoyanov** [comment]
@blueorangutan package

File: server/src/com/cloud/vm/UserVmManagerImpl.java [hunk 1]
```diff
@@ -6100,9 +6101,18 @@ public UserVm restoreVMInternal(Account caller, UserVmVO vm, Long newTemplateId)
                     throw ex;
                 }
             }
-            TemplateDataStoreVO tmplStore = _templateStoreDao.findByTemplateZoneReady(template.getId(), vm.getDataCenterId());
-            if (tmplStore == null) {
-                throw new InvalidParameterValueException("Cannot restore the vm as the template " + template.getUuid() + " isn't available in the zone");
+
+            TemplateDataStoreVO tmplStore;
```
Comment (DaanHoogland) [inline server/src/com/cloud/vm/UserVmManagerImpl.java]: please, extract this bit as a separate private method, i.e. checkRestorePossible().

**Comment by borisstoyanov** [comment]
@blueorangutan test

**Comment by nvazquez** [comment]
Thanks @borisstoyanov and @DaanHoogland for reviewing

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by borisstoyanov** [comment]
@blueorangutan package

**Comment by borisstoyanov** [comment]
@blueorangutan test

**Comment by nvazquez** [comment]
Thanks @borisstoyanov @DaanHoogland for reviewing, should we re run tests?

**Comment by nvazquez** [comment]
Closing, as it is included on 2408

**Pull request closed without merging**